tests/baselines/reference/arraySlice.types

```diff
@@ -5,7 +5,7 @@ var arr: string[] | number[];
 >arr : string[] | number[]
 
 arr.splice(1, 1);
->arr.splice(1, 1) : string[] | number[]
+>arr.splice(1, 1) : (string | number)[]

```

Comment:
array types merged

Reply:
Is this change correct? When you have an array that contains either numbers or strings, then splicing shouldn't suddenly result in an array that can contain numbers AND strings.

Reply:
That's a very good question. This behavior is not new - the "carveout" I mentioned in summary of the pull was originally added in #53489 - 

> With this PR, when we see a type like `(A[] | B[])["member"]` and look for signatures on it, and find none, we reconstruct `(A | B)[]["member"]` and instead look up signatures on that. This only applies to array and tuple types because, well, arrays are special. We treat them as covariant when they're not, and we know their collection-like behavior is consistent. 

When @weswigham says "We treat them as covariant ..." I believe he is talking in the past tense - meaning that even before that pull #53489 there were already cases of such.  (to be continued .... I have to go).

Reply:
@MartinJohns - Now returns the narrower type as you prefer.